File: app/src/main/java/com/bth/running/coaching/Coach.java
package com.bth.running.coaching;

import com.bth.running.weather.model.Weather;

import rx.Observable;

/**
 * @author Martin Macheiner
 *         Date: 20.09.2017.
 */

public interface Coach {

    Observable<Weather> getWeatherForecast(String place);

    void setUserBodyInformation(int height, double weight);

    double getUserWeight();

    int getUserHeight();

}

File: app/src/main/java/at/shockbytes/corey/running/coaching/Coach.kt
package at.shockbytes.corey.running.coaching


import at.shockbytes.corey.running.weather.model.Weather
import io.reactivex.Observable

/**
 * @author Martin Macheiner
 * Date: 20.09.2017.
 */

interface Coach {

    val userWeight: Double

    val userHeight: Int

    fun getWeatherForecast(place: String): Observable<Weather>

    fun setUserBodyInformation(height: Int, weight: Double)

}
